posthog/api/insight.py

```diff
@@ -100,12 +100,29 @@ def log_insight_activity(
         )
 
 
+class DashboardsField(serializers.Field):
+    def get_attribute(self, instance):
+        # We pass the object instance onto `to_representation`,
+        # not just the field attribute.
+        return instance
+
+    def to_representation(self, insight: Insight) -> List[int]:
+        return [tile.dashboard.id for tile in insight.dashboard_tiles.all()]
+
+    def to_internal_value(self, data) -> List[Dashboard]:
+        return list(Dashboard.objects.filter(id__in=data))
+
+
 class InsightBasicSerializer(TaggedItemSerializerMixin, serializers.ModelSerializer):
     """
     Simplified serializer to speed response times when loading large amounts of objects.
     """
 
     created_by = UserBasicSerializer(read_only=True)
+    dashboards = DashboardsField(
```

Comment:
Why not `dashboard_ids`? `dashboards` to me suggests serialization with `DashboardSerializer`